### PR TITLE
Add delete_function API

### PR DESCRIPTION
**Pull request by serena-ruan**
Add delete_function API for clients.

### CODE REVIEW

File: src/ucai/core/databricks.py [hunk 1]
```diff
@@ -602,6 +602,22 @@ def _execute_uc_functions_with_serverless(
...
+        full_function_name: str,
+        force: Optional[bool] = None,
+    ) -> None:
+        """
+        Delete a function by its full name.
+
+        Args:
+            full_function_name: The full name of the function to delete.
+                It should be in the format of "catalog.schema.function_name".
+            force: Force deletion even if the function is notempty. Defaults to None.
```
Comment (harupy) [inline src/ucai/core/databricks.py]: what does None mean?

Comment (harupy) [reply src/ucai/core/databricks.py]:
```suggestion
            force: Force deletion even if the function is not empty. Defaults to None.
```

File: src/ucai/core/client.py [hunk 2]
```diff
@@ -163,6 +163,20 @@ def _execute_uc_function(
         Internal logic for executing a UC function.
         """
 
+    @abstractmethod
+    def delete_function(
+        self,
+        full_function_name: str,
```
Comment (harupy) [inline src/ucai/core/client.py]: do we use `full_function_name` in other methods? We should be consistent.

File: src/ucai/core/databricks.py (hunk 1 above)
Comment (serena-ruan) [reply src/ucai/core/databricks.py]: Honestly I don't know :) The underlying sdk uses default value None, if it's none then it doesn't pass this field to the request body

File: src/ucai/core/client.py [hunk 3]
```diff
@@ -163,6 +163,20 @@ def _execute_uc_function(
...
+        self,
+        function_name: str,
+        **kwargs,
+    ) -> None:
+        """
+        Delete a function by its full name.
+
+        Args:
+            function_name: The full name of the function to delete.
+                It should be in the format of "catalog.schema.function_name".
```
Comment (harupy) [inline src/ucai/core/client.py:177]:
```suggestion
                It should be in the format of "catalog.schema.function_name".
            kwargs: ...
```

to explain what kwargs is.

File: src/ucai/core/databricks.py (hunk 1 above)
Comment (harupy) [reply src/ucai/core/databricks.py]: > The underlying sdk uses default value None

I think we can mention this if the underlying client decides what to do with None.

**Review by harupy — APPROVED** [review]
LGTM

**Pull request merged**